### ISSUE TITLE
Heterogeneity not calculable from "ic" method 

### ISSUE THREAD

**Issue by cesterwo1056** [psychmeta/psychmeta#94]
Sorry if this is a foolish question but I'm getting an error when I try to get heterogeneity from an analysis when I run a meta-analysis with individual correction and collapse dependencies within the data. 

Here is what I'm using: 

```
out<-ma_r(
 rxyi="Construct_Corr", 
 n="N", 
 data = data, 
 sample_id = Study, 
 check_dependence = TRUE, 
 collapse_method = c("average"),
 conf_level =.95, 
 ryy = "Reliability_Y", 
 construct_x = 'construct_x', 
 construct_y = 'construct_y', 
 moderators = 'moderators', 
 ma_method = "ic", 
 clean_artifacts = TRUE, 
 impute_artifacts = TRUE
)
```

From this, I get a good summary output with the information I need but when I try to use that `out` object with heterogeneity I get the following error:

```
het<-heterogeneity(out)
```

```
Error in if (f_low * f_upp > 0) tau_squared_r <- 0 else tau_squared_r <- uniroot(f,  : 
  missing value where TRUE/FALSE needed
```

**Comment by cesterwo1056** [comment]
Attempted this again and had the same general result but, it worked for one dataframe and not the other.

```
out<-ma_r( rxyi = "r", n = "N", sample_id = "Study", 
ma_method = c("ic"), 
correction_method = "auto", wt_type = c("sample_size"), 
collapse_method = c("average"),
correct_rel = TRUE,
correct_rxx = TRUE, correct_ryy = TRUE, correct_rr = FALSE, 
correct_rr_x = FALSE, correct_rr_y = FALSE, 
indirect_rr = FALSE, indirect_rr_x = FALSE, 
indirect_rr_y = FALSE, rxx = "rxx", 
rxx_restricted = FALSE, rxx_type = "alpha", 
ryy = "ryy", ryy_restricted = FALSE, 
ryy_type = "alpha", 
data = data)
```

This returned a correct summary of the analysis via `summary(out)` but, when I go to run `heterogeneity(out)` I receive the following:

```
NaNs producedNaNs producedError in if (f_low * f_upp > 0) tau_squared_r <- 0 else tau_squared_r <- uniroot(f,  : 
  missing value where TRUE/FALSE needed
```

Investigation on forums seem to indicate that this is a uniroot issue or something with a function but I can't say for sure since I'm just a user.

**Comment by cesterwo1056** [comment]
The error seems focused around the ```rxx```  and ```ryy``` variables.

I'm able to run the following and get heterogeneity:

```
out<-ma_r( rxyi = "r", n = "N", sample_id = "Study", 
ma_method = c("ic"), 
correction_method = "auto", wt_type = c("sample_size"), 
collapse_method = c("average"),
#correct_rel = TRUE,
#correct_rxx = TRUE, 
#correct_ryy = TRUE, 
#correct_rr = FALSE, 
#correct_rr_x = FALSE, 
#correct_rr_y = FALSE, 
#indirect_rr = FALSE, 
#indirect_rr_x = FALSE, 
#indirect_rr_y = FALSE, 
#rxx = "rxx", 
#rxx_restricted = FALSE, 
#rxx_type = "alpha", 
#ryy = "ryy", 
#ryy_restricted = FALSE, 
#ryy_type = "alpha", 
data = Openness_x_Acceptance)
```

My data has ```rxx``` and ```ryy``` for 3 different studies but not for the remaining 4 studies in question. 

I'm not sure if this helps or not though.

**Comment by cesterwo1056** [comment]
I wonder if this is something to do with the imutation of reliabilty. 

I noticed that when I do a forrest plot after running the earlier code that triggers a heterogeneity error that I am missing data for the studies that didn't report reliablity.....

**Comment by jadahlke** [comment]
Hi Connor - can you please send us a data file so that we can reproduce the error? We'll need a complete reproducible example (code + inputs) to troubleshoot this. You don't need to post it here, email is fine (issues@psychmeta.com).

**Comment by cesterwo1056** [comment]
> 
> 
> Hi Connor - can you please send us a data file so that we can reproduce the error? We'll need a complete reproducible example (code + inputs) to troubleshoot this. You don't need to post it here, email is fine ([issues@psychmeta.com](mailto:issues@psychmeta.com)).

Thanks for getting back to me. I'll shoot over a zip with code and data later on today.

**Comment by cesterwo1056** [comment]
Seems the email didn't send and is kicking back so here is the zip in case I can't get it to work a different way. 


[Error_Reporduction.zip](https://github.com/psychmeta/psychmeta/files/4982345/Error_Reporduction.zip)

**Comment by bwiernik** [comment]
Thanks. The bug has to do with the missing artifact information for some of the studies and the imputation not working as expected. We'll get this fixed.

**Comment by cesterwo1056** [comment]
Awesome! 👍 Wish I was knowledgeable enough to help with that. 
I'm very relieved it wasn't me doing something stupid by accident though 😌

**Comment by bwiernik** [comment]
This is fixed in psychmeta 2.4.2, available now from `remotes::install_github("psychmeta/psychmeta")`
